spec/units/prompter_spec.rb

```diff
@@ -166,6 +166,22 @@
           "#{File.basename(Dir.getwd.sub(/\A#{Dir.home}/, '~'))} "
         )
       end
+
+      it 'replaces %g with the absolute path of the current git binary' do
+        env = env_double(format: '%g')
+        prompter = Gitsh::Prompter.new(env: env)
+
+        expect(prompter.prompt).to eq "#{env.git_command} "
+      end
+
+      it 'replaces %G with the basename of the current git binary' do
+        env = env_double(format: '%G')
```

Comment:
What do you think about stubbing `git_command` here, so that we can make the assertion more explicit? e.g. if we stubbed it as `/usr/local/bin/my-custom-git`, we could verify the output was `my-custom-git`.

Reply:
More than happy to, as I rather like the idea of a more explicit test. Not sure how to stub it—would I do that with the `env_double` below or somewhere else?

Reply:
In `#env_double`, I think you could drop `allow(env).to receive(:git_command).and_return(…)` and instead add `git_command: …` to the `default_attrs` Hash. That would allow you to override the value of `git_command` by calling `env_double(git_command: 'some custom value')`

Reply:
Cool. Having never worked with ruby before, it's remarkably expressive.